core/modules/replica/DatabaseServices.h

```diff
@@ -102,7 +103,7 @@ class DatabaseServices
      *
      * The Job::Option object is explicitly passed as a parameter to avoid
      * making a call back to the job which may create a deadlock because
-     * method saveState() is meant to be called by the jobs' implementations.
+     * method saveState() is meant to be called by the job's implementations.
```

Comment:
it strikes me as a little strange that there are overloads of `saveState` that each have different calling requirements (functions with different API should be named differently? ...maybe they should be named differently, `saveControllerState` and `saveJobState`?)

Reply:
That's a reason why C++ has overloading :-)  A contract of a method is defined by its name and a collection (and types) of its parameters.